From 58b01dac64ff491f74c989da33b1b261ca196a0f Mon Sep 17 00:00:00 2001
From: Peleg Vigodny <pvigodny@habana.ai>
Date: Wed, 04 Jan 2023 11:20:47 +0200
Subject: [PATCH] [SW-117471] erom: tpm init: fix dev wait condition

Before starting TPM initialization, we call tpm_tis_wait_init() to wait
for SPI communication with TPM to start.

We assume this happens once we are able to read locality reg and see
bit 7 (TPM_ACCESS_VALID) is on.

However, it seems that while TPM device is coming up and not yet ready
to communicate, we effectively read 0xFFs from the locality reg - this
means our bit 7 check passes whereas the TPM is not really ready for
communication and might start tpm initialization too early.

This change waits for a value other than 0xFF to be read before going
on with TPM initialization.

Change-Id: Id6428cfba8244616d60b1369cf41e0b27ae8b729
Signed-off-by: Peleg Vigodny <pvigodny@habana.ai>
---

diff --git a/arch/gaudi2/common/tpm2.c b/arch/gaudi2/common/tpm2.c
index 9388c42..00b31e2 100644
--- a/arch/gaudi2/common/tpm2.c
+++ b/arch/gaudi2/common/tpm2.c
@@ -1000,7 +1000,7 @@
 		else if (ret != 0)
 			break;
 
-		if (status & TPM_ACCESS_VALID)
+		if (status & TPM_ACCESS_VALID && status != 0xff)
 			return 0;
 	} while (!timeout_expired(start, MS_TO_US(stop)));
 
